### PR TITLE
Stops plants from growing when they do not have enough resources

### PR DESCRIPTION
**Pull request by Brandon-Huu**
## About the PR

Plants were growing without the need for water or nutrients.
Same as the closed pull request https://github.com/space-wizards/space-station-14/pull/21502 but with the branch separate from master.
## Why / Balance

This issue was brought up in https://github.com/space-wizards/space-station-14/issues/21494 .

Plants should not grow without the resources needed to build their cell structures.

## Technical details

Checks if a plant is missing nutrients or water every update instead of waiting for `_random.Prob(0.35f)` to return true.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: Brandon_H
- fix: Plants can no longer grow without sufficient water and nutrients.

### CODE REVIEW

File: Content.Server/Botany/Systems/PlantHolderSystem.cs [hunk 1]
```diff
@@ -441,29 +441,24 @@ public void Update(EntityUid uid, PlantHolderComponent? component = null)
             component.Health -= 6 * healthMod;
         }
 
-        // Make sure the plant is not starving.
-        if (_random.Prob(0.35f))
+        if (component.SkipAging < 10)
```
Comment (Partmedia) [inline Content.Server/Botany/Systems/PlantHolderSystem.cs:444]: Why was this added here?

File: Content.Server/Botany/Systems/PlantHolderSystem.cs [hunk 2]
```diff
@@ -441,29 +441,24 @@ public void Update(EntityUid uid, PlantHolderComponent? component = null)
...
+                component.Health += Convert.ToInt32(_random.Prob(0.35f)) * healthMod;
             }
             else
             {
                 AffectGrowth(uid, -1, component);
                 component.Health -= healthMod;
             }
 
-            if (component.DrawWarnings)
-                component.UpdateSpriteAfterUpdate = true;
```
Comment (Partmedia) [inline Content.Server/Botany/Systems/PlantHolderSystem.cs:458]: Why was this removed?

**Review by Partmedia — COMMENTED** [review]
Thank you for your investigation and contribution. Could you please respond to the review comments?

File: Content.Server/Botany/Systems/PlantHolderSystem.cs (hunk 2 above)
Comment (Brandon-Huu) [reply Content.Server/Botany/Systems/PlantHolderSystem.cs:458]: It was removed because there is a duplicate within the same if block on line 469-470.

File: Content.Server/Botany/Systems/PlantHolderSystem.cs (hunk 1 above)
Comment (Brandon-Huu) [reply Content.Server/Botany/Systems/PlantHolderSystem.cs:444]: I assumed that the random probability was added to prevent `component.SkipAging` from getting too high.

Because of that assumption, I made it so `component.SkipAging` doesn't go above 10 solely due to insufficient water or nutrients

Comment (Partmedia) [reply Content.Server/Botany/Systems/PlantHolderSystem.cs:444]: I think this would be a great thing to explain in the comments. Also note the extraneous whitespace on line 446.

File: Content.Server/Botany/Systems/PlantHolderSystem.cs (hunk 2 above)
Comment (Partmedia) [reply Content.Server/Botany/Systems/PlantHolderSystem.cs:458]: Aha, this wasn't clear to me just looking at the diff. Looking at the change locally, this makes sense.

**Comment by Partmedia** [comment]
A nitpick about the CL entry:
"You now need water and nutrients to grow plants in hydroponic trays." is not ideal because this has always been the intention and doesn't explain what was fixed. Something like "Plants can no longer grow without sufficient water and nutrients." better explains what was fixed.

**Review by Partmedia — APPROVED** [review]
Thanks for the fix. I hope you'll update the comments to better explain the change on line 444.

**Comment by Partmedia** [comment]
Thank you for your contribution. You were missing a period at the end of your Changelog. Please review the [Changelog guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html#changelog).
